[fix] In the case of non-DimensionedTensorTypes, default to checking if either CPU or GPU fusion is enabled

bwasti opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22658 [fix] In the case of non-DimensionedTensorTypes, default to checking if either CPU or GPU fusion is enabled**

Differential Revision: [D16209907](https://our.internmc.facebook.com/intern/diff/D16209907)

bwasti committed:
[fix] In the case of non-DimensionedTensorTypes, default to checking if either CPU or GPU fusion is enabled

jamesr66a commented on test/test_jit_fuser.py:
```diff
@@ -895,6 +895,28 @@ def my_broadcasted_cell(a, b, c):
+
+        R_in = torch.ones(2, 2)
+        v_in = torch.ones(2)
+        g_str = str(cholupdate_no_print.graph_for(R_in, v_in))
```
Can we use FileCheck instead here?

jamesr66a replied:
Also, is this ensured to only test on CPU?

jamesr66a commented on torch/csrc/jit/passes/graph_fuser.cpp:
```diff
@@ -192,6 +195,9 @@ struct GraphFuser {
     }
+    for (const auto& input : node->inputs()) {
+      fusableDevice &= isFusableDevice(input);
+    }
```
This is all getting way too complicated. Can you write out a comment describing the logic of when a FusionGroup is or isn't emitted?

bwasti replied:
oh whoops, didn't mean to add that. it isn't necessary

bwasti committed:
Update on "[fix] In the case of non-DimensionedTensorTypes, default to checking if either CPU or GPU fusion is enabled"

[fix] In the case of non-DimensionedTensorTypes, default to checking if either CPU or GPU fusion is enabled

gh-metadata: pytorch pytorch 22658 gh/bwasti/25/head
